src/coreclr/tools/aot/ILCompiler.Build.Tasks/ComputeManagedAssembliesToCompileToNative.cs

```diff
@@ -122,12 +122,30 @@ public override bool Execute()
                     continue;
                 }
 
-                // Remove any assemblies whose implementation we want to come from CoreRT's package.
-                // Currently that's System.Private.* SDK assemblies and a bunch of framework assemblies.
-                if (coreRTFrameworkAssembliesToUse.Contains(Path.GetFileName(itemSpec)))
+                var assemblyFileName = Path.GetFileName(itemSpec);
+                if (assemblyFileName == "WindowsBase.dll")
                 {
-                    assembliesToSkipPublish.Add(taskItem);
-                    continue;
+                    // There two instances of WindowsBase.dll, one small one, in the NativeAOT framework
+                    // and real one in WindowsDesktop SDK. We want to make sure that if both are present,
+                    // we will use ne from WindowsDesktop SDK, and not from NativeAOT framework.
+                    foreach (ITaskItem taskItemToSkip in FrameworkAssemblies)
+                    {
+                        if (Path.GetFileName(taskItemToSkip.ItemSpec) == assemblyFileName)
+                        {
+                            assembliesToSkipPublish.Add(taskItemToSkip);
+                            continue;
```

Comment:
I think this should be `break;` and the `continue;` should be after the for-loop.